syft/pkg/cataloger/deb/cataloger.go

```diff
@@ -68,6 +68,8 @@ func (c *Cataloger) Catalog(resolver source.FileResolver) ([]pkg.Package, []arti
 
 			// fetch additional data from the copyright file to derive the license information
 			addLicenses(resolver, dbLocation, p)
+
+			p.SetID()
```

Comment:
Maybe a comment for context that THIS is where the package become immutable?

I think it's going to be hard to enforce this by convention going forward so I'm trying to think of a better way than a giant warning comment.

Reply:
Agreed. I think the need for this `SetID` method is the only thing in this PR that seems like it needs continued thought and follow-up — in a **_future_** PR. I can see this getting difficult to enforce and reason about for future developers, and it'd be great to find ways to reduce the burden on catalogers and achieve more immutability. But I think the overall design here is definitely an improvement. 💯